compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -862,6 +865,20 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
     ) -> BlockAnd<()> {
         let span = source_info.span;
 
+        let opt_break_index = self
+            .scopes
+            .const_continuable_scopes
+            .iter()
+            .rposition(|const_continuable_scope| const_continuable_scope.region_scope == scope);
+
+        let Some(break_index) = opt_break_index else {
+            if self.scopes.const_continuable_scopes.is_empty() {
+                self.tcx.dcx().emit_fatal(ConstContinueNotInLoopMatch { span })
+            } else {
+                self.tcx.dcx().emit_fatal(ConstContinueInvalidJump { span })
+            }
+        };
```

Comment:
nit: this can be in an `unwrap_or_else` (I think it'd be slightly more readable)